### DISCUSSION TITLE
Laravel 8: laravel-csp package installation error

### DISCUSSION THREAD

**Discussion by sys-auditing** [spatie/laravel-csp#94] [Q&A]
Hi,

i'm getting these errors when trying to install spatie/laravel-csp via running the command ```composer require spatie/laravel-csp ``` 

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires spatie/laravel-csp ^2.7 -> satisfiable by spatie/laravel-csp[2.7.0].
    - spatie/laravel-csp 2.7.0 requires spatie/laravel-package-tools ^1.11 -> found spatie/laravel-package-tools[1.11.0, 1.11.1, 1.11.2, 1.11.3] but the package is fixed to 1.6.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

How can I solve the issue?

Thank you

**Comment by mfn** [comment]
> `Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.`

did you try this?

Culprit seems to be this:
> `but the package is fixed to 1.6.2 (lock file version) by a partial update and that version does not match`

If the above does not work, you can try (but pls make a backup first) removing the lockfile and update all packages etc.

If nothing works, please share your composer.json and, in your case, composer.lock.

**Comment by sys-auditing** [comment]
@mfn , yes i have tried adding the option ```--with-all-dependencies ```, but it's gave me a lot of errors.
So i decided to install an old version 2.6.4

Thank you
